packages/veto-app/frontend/src/components/molecules/ReceiveForm/ReceiveForm.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import { Input, Label, Text, Box, Select, Flex, Button } from '@veriphi/veto-ui'
+import useNewAddress from '../../../hooks/useNewAddress'
+import ErrorMessage from '../../atoms/ErrorMessage'
+
+const ReceiveBTC = () => {
+  const { address, addressError, isValidating } = useNewAddress()
+
+  return (
+    <Box>
+      <Text variant="heading2">Receive BTC</Text>
+
+      <Label>Receving Wallet</Label>
+      <Select>
+        <option>Core Wallet</option>
+      </Select>
+      <Label>Label</Label>
+      <Input placeholder="Internal label for this transaction" />
+
+      <Label>Address</Label>
+
+      <Flex>
+        {addressError ? (
```

Comment:
We can discuss this but imo multi-ternary should be avoided. If the logic is complex enough to require an `else if` its best to offload it to a function|component or assign the resulting component to a variable before the `return`.
```
function renderAddress(address:Error, isValidating, addressError){
  if(addressError){
    return <ErrorMessage>{addressError}</ErrorMessage>
  } else if(isValidating){
    return '...Fetching new address...';
  }

  return (<>
    <Text>{address}</Text>
     <Button>copy to clipboard</Button>
   </>)
}
```

```
<Flex>
  {renderAddress(address, isValidating, addressError)}
</Flex>
```

Or as a component
```
function Address(address:Error, isValidating, addressError){
  if(addressError){
    return <ErrorMessage>{addressError}</ErrorMessage>
  } else if(isValidating){
    return '...Fetching new address...';
  }

  return (<>
    <Text>{address}</Text>
    <Button>copy to clipboard</Button>
   </>)
}
```

```
<Flex>
  <Address 
    address={address}
    isValidating={isValidating}
    addressError={addressError}
  />
</Flex>
```

Reply:
I'll go with option 1 :) I like